packages/backend/src/apps/lettersg/common/api.ts

```diff
@@ -0,0 +1,34 @@
+export enum LetterSgEnvironment {
+  Staging = 'test',
+  Prod = 'live',
+}
+
+const STAGING_ENV_API_KEY_PREFIX = 'test_'
+const STAGING_ENV_BASE_URL = 'https://staging.letters.gov.sg/api'
+
+// TODO (mal): test this when prod api is out
+const PROD_ENV_API_KEY_PREFIX = 'live_'
+const PROD_ENV_BASE_URL = 'https://letters.gov.sg/api'
+
+export function getEnvironmentFromApiKey(apiKey: string): LetterSgEnvironment {
+  if (apiKey.startsWith(PROD_ENV_API_KEY_PREFIX)) {
+    return LetterSgEnvironment.Prod
+  }
+
+  if (apiKey.startsWith(STAGING_ENV_API_KEY_PREFIX)) {
+    return LetterSgEnvironment.Staging
+  }
+
+  throw new Error('LetterSG API key has unrecognized prefix!')
```

Comment:
nit: might there be a more specific error type in the codebase?

Reply:
for now, this is just a straightforward error message, so our codebase just throws a standard error